testing/trino-testing/src/main/java/io/trino/testing/QueryAssertions.java

```diff
@@ -335,7 +335,7 @@ public static void copyTable(QueryRunner queryRunner, QualifiedObjectName table,
     {
         long start = System.nanoTime();
         log.info("Running import for %s", table.getObjectName());
-        @Language("SQL") String sql = format("CREATE TABLE %s AS SELECT * FROM %s", table.getObjectName(), table);
+        @Language("SQL") String sql = format("CREATE TABLE IF NOT EXISTS %s AS SELECT * FROM %s", table.getObjectName(), table);
```

Comment:
When i suggested this, i thought about this being opt-in behavior. What was your thinking about this?

Reply:
It does not hurt. If table exists it is skipped. If not it is created. The only problem is when table exists and it is different than expected. However we only use for TPCH tables so they are quite immutable. Notice that also we don't reuse anything so table usually do not exists.

Reply:
but there could be an initialization race, of something.
if this is unconditionally enabled, let's eg validate row count of the table

Reply:
> but there could be an initialization race, of something.

Do you mean that this could be loaded in multiple threads? Like single environment used for several unit tests, then I guess one them should be failed as it won't be able to commit CTAS.

Reply:
Threads, or previous startup was aborted. Or there is a rogue test which does modify one of these tables.

Reply:
> Threads, or previous startup was aborted.

As I said, that should be fine as only one thread can successfully finish. If build was aborted, transaction is aborted and next build can continue and create missing tables.

> Or there is a rogue test which does modify one of these tables.

We would know about that rouge tests already, unless it is run in isolation (like single tests per environment).


Notice, that reusing will be most of the time disabled (like 99% or more), so nothing changes. Reusing is like `--attach` for product tests launcher. We cannot promise anything, if you reuse you are on your own. Things may work or not depending on what you do and how tests are implemented. You might gain time or lose time due some issues.